starfish/test/full_pipelines/api/test_iss.py

```diff
@@ -6,15 +6,27 @@
 import sys
 import tempfile
 import unittest
+from typing import Sequence
+
+import jsonpath_rw
 
 from starfish.util import clock
 
 
-def get_codebook(tempdir):
-    json_filepath = os.path.join(tempdir, "formatted", "experiment.json")
+def get_jsonpath_from_file(json_filepath_components: Sequence[str], jsonpath: str):
+    """
+    Given a series of filepath components, join them to find a file <FILE>.  Open that file, and locate a specific value
+    in the json structure <PATH>.  Join the directory path of <FILE> and <PATH> and return that.
+
+    For example, if json_path_components is ["/tmp", "formatted", "experiment.json"] and jsonpath is
+    "$['hybridization']", this method will open /tmp/formatted/experiment.json, decode that as a json document, and
+    locate the value of the key 'hybridization'.  It will return /tmp/formatted/XXX, where XXX is the value of the key.
```

Comment:
This seems more complex than simply making a user pass the codebook location separately as a string. Could you explain more why you're doing it this way?

Reply:
Nevermind.. looking at the next PR now.